### PR TITLE
Add RegExp option for hierarchySeparator

### PR DESCRIPTION
**Pull request by usulpro**
Issue: #151

`hierarchySeparator` option allows only a string with RegExp

## What I did

`hierarchySeparator` could take a `Regular Expression` directly (as well as strings)

## How to test

run `cra-kitchen-sink`

### CODE REVIEW

File: addons/options/src/preview/index.js [hunk 1]
```diff
@@ -7,14 +7,26 @@ export function init() {
   // NOTE nothing to do here
 }
 
+function regExpStringify(exp) {
+  if (!exp) return null;
```
Comment (shilman) [inline addons/options/src/preview/index.js]:
```js
if(typeof exp === 'string') return exp;
if(Object.prototype.toString.call(exp) === '[object RegExp]') return exp.source;
return null;
```

**Review by shilman — CHANGES_REQUESTED** [review]

**Review by shilman — APPROVED** [review]

File: examples/cra-kitchen-sink/src/stories/index.js [hunk 2]
```diff
@@ -266,7 +266,7 @@ storiesOf('component.Button', module)
 
 // Atomic
 
-storiesOf('Cells¯\\_(ツ)_/¯Molecules.Atoms/simple', module)
+storiesOf('Cells/Molecules.Atoms/simple', module)
```
Comment (igor-dv) [inline examples/cra-kitchen-sink/src/stories/index.js]: :cry:  ok..

File: addons/options/src/preview/index.js [hunk 3]
```diff
@@ -7,14 +7,25 @@ export function init() {
...
+export function setOptions(newOptions) {
   const channel = addons.getChannel();
   if (!channel) {
     throw new Error(
       'Failed to find addon channel. This may be due to https://github.com/storybooks/storybook/issues/1192.'
     );
   }
+  const options = {
+    ...newOptions,
+    hierarchySeparator: regExpStringify(newOptions.hierarchySeparator),
```
Comment (igor-dv) [inline addons/options/src/preview/index.js]: what will happen if null will be passed here ?

**Review by igor-dv — APPROVED** [review]
cc @joscha

File: addons/options/src/preview/index.js (hunk 3 above)
Comment (UsulPro) [inline addons/options/src/preview/index.js]: it will be ignored

**Comment by UsulPro** [comment]
so we can merge it if nobody objects?

**Pull request merged**